platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConverterService.java

```diff
@@ -216,11 +203,20 @@ public boolean isInitialized() {
         return initialized;
     }
 
-    @SuppressWarnings("unchecked")
     @Nullable
     <T> ConfigConverter<T> getConverterForType(@NonNull final Class<T> valueType) {
         throwIfNotInitialized();
         Objects.requireNonNull(valueType, "valueType must not be null");
-        return (ConfigConverter<T>) converters.get(valueType);
+        return getOrAdConverter(valueType);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> ConfigConverter<T> getOrAdConverter(Class<T> valueType) {
```

Comment:
Let's add a short doc regarding the special case of the enum converter to this method

Reply:
Done